src/nvector/pthreads/nvector_pthreads.c

```diff
@@ -3436,14 +3597,16 @@ int N_VLinearCombinationVectorArray_Pthreads(int nvec, int nsum, sunrealtype* c,
     if (nsum == 1)
     {
       N_VScale_Pthreads(c[0], X[0][0], Z[0]);
-      return (0);
+      SUNCheckLastErr();
+      return SUN_SUCCESS;
     }
 
     /* should have called N_VLinearSum */
     if (nsum == 2)
     {
       N_VLinearSum_Pthreads(c[0], X[0][0], c[1], X[1][0], Z[0]);
-      return (0);
+      SUNCheckLastErr();
+      return SUN_SUCCESS;
     }
 
     /* should have called N_VLinearCombination */
```

Comment:
Below add `SUNAssert` on `malloc`, `SUNCheckCall` on `N_VLinearCombination_Pthreads`, remove `retval` and return `SUN_SUCCESS`.

Further below, `SUNAssert` on `malloc`, `SUNCheckCall` on `N_VScaleVectorArray_Pthreads` and `N_VLinearSumVectorArray_Pthreads`, remove `retval` and return `SUN_SUCCESS`.